python/ml4ir/base/data/tfrecord_helper.py

```diff
@@ -42,12 +42,16 @@ def get_example_proto(row, features):
     """
 
     features_dict = dict()
-
     for feature_info in features:
         feature_name = feature_info["name"]
         feature_fn = _get_feature_fn(feature_info["dtype"])
-        features_dict[feature_name] = feature_fn([row[feature_name]])
-
+        # When applying functions with axis=1, pandas performs upcasting,
+        # so if we have a mix of floats/ints, converts everything to float
+        # that breaks this part of the code. Example:
+        # https://stackoverflow.com/questions/47143631/
+        # how-do-i-preserve-datatype-when-using-apply-row-wise-in-pandas-dataframe
+        value_upcasted = row[feature_name].astype(feature_info["dtype"])
```

Comment:
I documented inline the change.. Essentially, if you have `float` and `int` in a row, the row is converted to floats. This results in failure here, as the protos use the type in the config.

Reply:
Can you add a `FIXME` tag perhaps? But yeah, looks like there might not be a fix?

Reply:
I've not had any issue with int and floats though.

Reply:
There is nothing to fix here. Line 53 is the fix. I had the issue and spent quite some time to debug. Perhaps different Pandas versions..